src/app/services/auth.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+import { ApiService } from './api.service';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  private _isLoggedIn$ = new BehaviorSubject<boolean>(false);
+  isLoggedIn$ = this._isLoggedIn$.asObservable();
+  key:string='';
+  constructor(private apiService: ApiService) {
+    this._isLoggedIn$.next(this.isAuthenticated());
+  }
+
+ 
+
+  private isAuthenticated(): boolean {
+    const username = localStorage.getItem('loggedInUser');
+    return!!username && localStorage.getItem(username)!== null;
+  }
+
+  login(username: string, password: string): Observable<boolean> {
+    return this.apiService.login(username, password).pipe(
+      map(response => {
+        if (response) {
+          localStorage.setItem('loggedInUser', username);
+          const token = response;
+          localStorage.setItem('auth', token);
+          this._isLoggedIn$.next(true);
+          return true;
+        } else {
+          return false;
```

Comment:
response need to be proper object
note : implement base request and inherite all request from it. also define response objects as well